obs/project.go

```diff
@@ -91,3 +96,110 @@ type RepositoryPath struct {
 	Project    string `json:"project" xml:"project,attr"`
 	Repository string `json:"repository" xml:"repository,attr"`
 }
+
+type Status struct {
+	XMLName xml.Name `json:"status" xml:"status"`
+	Code    string   `json:"code" xml:"code,attr"`
+	Summary string   `json:"summary" xml:"summary"`
+}
+
+type APIError struct {
+	HTTPStatusCode int
+	XMLStatusCode  string
+	Message        string
+}
+
+func (e *APIError) Error() string {
+	return fmt.Sprintf("HTTP status %d: %s (%s)", e.HTTPStatusCode, e.XMLStatusCode, e.Message)
+}
+
+func (c Client) CreateProject(project *Project) error {
+	xmlData, err := xml.MarshalIndent(project, "", " ")
+	if err != nil {
+		return err
+	}
+
+	urlPath := path.Join(c.APIURL, "source", project.Name, "_meta")
```

Comment:
Please pay some more attention to the StackOverflow link that I shared earlier: https://stackoverflow.com/questions/34668012/combine-url-paths-with-path-join

> When I use `path.Join("http://foo", "bar")`, I get `http:/foo/bar`.

Which means that we can't use `path.Join` for joining URLs.

The accepted answer states:

> ...
> Use the [url.JoinPath](https://pkg.go.dev/net/url@go1.19beta1#JoinPath) function in Go 1.19 or later:
> ...

We use [Go 1.21](https://github.com/kubernetes-sigs/release-sdk/blob/main/go.mod#L3) at this time so using this function is acceptable.